algo-docker.sh

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+
+set -eEo pipefail
+
+ALGO_DIR="$(dirname "${0}")"
+
+usage() {
+    retcode="${1:-0}"
+    echo "To run algo from Docker:"
+    echo ""
+    echo "docker run --cap-drop ALL -it -v <path to configs>:/${ALGO_DIR}/configs -v <path to config.cfg>:/${ALGO_DIR}/config.cfg trailofbits/algo:latest"
+    echo ""
+    exit ${retcode}
+}
+
+if [ ! -d configs ] ; then
+  echo "Looks like you're not bind-mounting your configs into this container."
+  echo "This is something you want to do, or you'll lose your configurations"
+  echo "as soon as algo finishes."
+  echo ""
+  usage -1
+fi
+
+if [ ! -f config.cfg ] ; then
+  echo "Looks like you're not bind-mounting your config.cfg into this container."
+  echo "algo needs a configuration file to run."
+  echo ""
+  usage -1
+fi
+
+if [ ! -e /dev/console ] ; then
+  echo "Looks like you're trying to run this container without a TTY."
+  echo "If you don't pass `-t`, you can't interact with the algo script."
+  echo ""
+  usage -1
+fi
+
+# Ensure that `config.cfg` has the appropriate line endings
+tr -d '\r' < config.cfg > config.cfg.new
```

Comment:
Actually, turns out you can't, which is probably why I did it in this weird way in the first place:

When you do the in-place `sed`, it actually moves the file and tries to re-create it, which isn't possible because of the bind-mount of the single file. We can either put everything in a single directory and point `algo` at that, or go back to this weird way of doing things.